### PR TITLE
Add hardware description docs site

### PR DESCRIPTION
**Pull request by shamlian**
Add hardware overview with electrical and mechanical details and examples for connection to some single board computer development boards.

### CODE REVIEW

**Comment by raphaelcherney** [comment]
Personally, I feel like the Hardware description should come before the API docs. (explain "What is this" before? "How do I use this?")

**Comment by alsora** [comment]
silly nitpick: it's better if after a `.` you go to a new line.
This makes the pages easier to read on github or using an editor (because lines are shorter) while not affecting the rendered document

**Comment by shamlian** [comment]
> silly nitpick: it's better if after a `.` you go to a new line. This makes the pages easier to read on github or using an editor (because lines are shorter) while not affecting the rendered document

As you wish.

File: docs/hw/overview.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+The locating pins on the top cover are used to install and retain the faceplate.
+
+There are two cable passthroughs.
+The rear passthrough is good for quick prototyping while the one that penetrates the top cover and faceplate is useful to keep wires within the radius of the robot.
+
+Also visible with the faceplate removed is the adapter board, which is used to interface to external computers either through Bluetooth® or via USB-C®.
+More information on this board is available on the [Electrical](../electrical/) page.
+
+![Create® 3 from a bottom view, with the cargo bay removed.](data/bottom.jpg "Robot Bottom")
+The bottom of the robot includes four cliff sensors to keep the robot on solid ground, a caster (by default, the robot's center of gravity is forward of the center axis), charging contacts, two wheels with current sensors and encoders, and an optical odometry sensor.
```
Comment (alsora) [inline docs/hw/overview.md:23]: we are not exposing wheel currents data to users, should we mention it? or maybe should we expose it?

Comment (shamlian) [reply docs/hw/overview.md:23]: IMHO we should expose it.

**Review by alsora — APPROVED** [review]

**Pull request merged**